### ISSUE TITLE
V2.0 unit issues

### ISSUE THREAD

**Issue by awe2** [uiucsn/astro_ghost#14]
Linux OS.CentOS; on fresh install of astro_ghost running the example script given at https://astro-ghost.readthedocs.io/en/latest/; 

```python
import os
import sys
from astro_ghost.PS1QueryFunctions import getAllPostageStamps
from astro_ghost.TNSQueryFunctions import getTNSSpectra
from astro_ghost.NEDQueryFunctions import getNEDSpectra
from astro_ghost.ghostHelperFunctions import getTransientHosts, getGHOST
from astropy.coordinates import SkyCoord
from astropy import units as u
import pandas as pd
from datetime import datetime

#we want to include print statements so we know what the algorithm is doing
verbose = 1

#download the database from ghost.ncsa.illinois.edu
#note: real=False creates an empty database, which
#allows you to use the association methods without
#needing to download the full database first
getGHOST(real=False, verbose=verbose)

#create a list of the supernova names, their skycoords, and their classes (these three are from TNS)
snName = ['SN 2012dt', 'SN 1998bn', 'SN 1957B']

snCoord = [SkyCoord(14.162*u.deg, -9.90253*u.deg, frame='icrs'), \
            SkyCoord(187.32867*u.deg, -23.16367*u.deg, frame='icrs'), \
            SkyCoord(186.26125*u.deg, +12.899444*u.deg, frame='icrs')]

snClass = ['SN IIP', 'SN', 'SN Ia']

# run the association algorithm!
# this first checks the GHOST database for a SN by name, then by coordinates, and
# if we have no match then it manually associates them.
hosts = getTransientHosts(snName, snCoord, snClass, verbose=verbose, starcut='gentle',ascentMatch=True)
```

Fails with Traceback (edited)

```python

Traceback (most recent call last):

File "/projectsastroghost/example_script.py", line 35, in <module>
       hosts = getTransientHosts(snName, snCoord, snClass, verbose=verbose, starcut='gentle',ascentMatch=True)
File "/user/.local/lib/python3.9/site-packages/astro_ghost/ghostHelperFunctions.py", line 640, in getTransientHosts
	   tempHost3 = findNewHosts(transientName_remaining, snCoord_remaining, snClass_remaining, verbose, starcut, ascentMatch, px, savepath)           
File "/user/.local/lib/python3.9/site-packages/astro_ghost/ghostHelperFunctions.py", line 807, in findNewHosts        
    host_DF, host_dict_nospace_postDLR, noHosts, GD_SN = chooseByDLR(path, host_gals_DF, transients, fn, host_dict_nospace, todo="r")  
File "/user/.local/lib/python3.9/site-packages/astro_ghost/DLR.py", line 295, in chooseByDLR dist,
    R = calc_DLR(ra_SN, dec_SN, ra_host, dec_host, r_a, r_b, host_df, band)   
File "/user/.local/lib/python3.9/site-packages/astro_ghost/DLR.py", line 120, in calc_DLR    
    hostCoord = SkyCoord(ra_host*u.deg, dec_host*u.deg, frame='icrs')
File "/user/.local/lib/python3.9/site-packages/astropy/coordinates/sky_coordinate.py", line 339, in __init__
    skycoord_kwargs, components, info = _parse_coordinate_data(
File "/user/.local/lib/python3.9/site-packages/astropy/coordinates/sky_coordinate_parsers.py", line 313, in _parse_coordinate_data       
    _components[frame_attr_name] = attr_class(arg, unit=unit)    
File "/user/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py", line 667, in __new__
    self = super().__new__(cls, angle, unit=unit, **kwargs)      
File "/user/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py", line 140, in __new__
    angle = [Angle(x, unit, copy=False) for x in angle]          
File "/user/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py", line 140, in <listcomp>
    angle = [Angle(x, unit, copy=False) for x in angle]          
File "/user/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py", line 142, in __new__
    return super().__new__(cls, angle, unit, dtype=dtype, copy=copy, **kwargs)      
File "/user/.local/lib/python3.9/site-packages/astropy/units/quantity.py", line 565, in __new__
    value._set_unit(value_unit)\n   
File "/user/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py", line 163, in _set_unit
    super()._set_unit(self._convert_unit_to_angle_unit(unit))    
File "/user/.local/lib/python3.9/site-packages/astropy/units/quantity.py", line 2118, in _set_unit
  raise UnitTypeError( astropy.units.core.UnitTypeError: Angle instances require units equivalent to 'rad', but no unit was given.
```

Temporary Solution (I'm sure this isn't the actual fix... ): 

change line 120 of: astro_ghost/DLR.py

```python
hostCoord = SkyCoord(ra_host*u.deg, dec_host*u.deg, frame='icrs')
```

to 

```python
hostCoord = SkyCoord(np.array([ra_host,])*u.deg, np.array([dec_host,])*u.deg, frame='icrs')
```

NOTE: It doesn't really make sense to me at all why this is a temporary fix? u.deg should work with both float and array types.

**Comment by alexandergagliano** [comment]
Thanks for posting, @awe2! What version of astropy are you running?

**Comment by awe2** [comment]
Howdy Alex, its the 2.0.0 release.

```python
>>>astro_ghost.__version__
'2.0.0'
```

**Comment by alexandergagliano** [comment]
astropy, not astro_ghost :)

**Comment by awe2** [comment]
apologies, 5.3
```python
>>> astropy.__version__
'5.3'
```

**Comment by alexandergagliano** [comment]
@awe2 Thanks for catching this, it looks like astropy v5.3 has deprecated some functionality that I relied on and I was testing on an earlier version. I've just re-released the code as version 2.0.1; a fresh install should fix things. Let me know if you're still running into problems!

**Comment by awe2** [comment]
Tested on CentOS astropy 5.3 and astro_ghost 2.0.2-- works!

**Comment by awe2** [comment]
Re-opening: astro_ghost V 2.0.2 and astropy V5.3  
```python
---------------------------------------------------------------------------
UnitTypeError                             Traceback (most recent call last)
Cell In[15], line 4
      1 # run the association algorithm!
      2 # this first checks the GHOST database for a SN by name, then by coordinates, and
      3 # if we have no match then it manually associates them.
----> 4 hosts = getTransientHosts(snName, snCoord, snClass, verbose=True, starcut='gentle', ascentMatch=True)

File ~/.conda/envs/torch2.0.0/lib/python3.9/site-packages/astro_ghost/ghostHelperFunctions.py:621, in getTransientHosts(transientName, snCoord, snClass, verbose, starcut, ascentMatch, px, savepath, GHOSTpath, redo_search)
    617 df_transients_remaining = df_transients[df_transients['Name'].isin(notFoundNames)]
    619 snCoord_remaining =  df_transients_remaining['snCoord'].values
--> 621 tempHost2, notFoundCoords = getDBHostFromTransientCoords(snCoord_remaining, GHOSTpath);
    622 found_by_coord = len(transientName) - len(notFoundCoords) - found_by_name
    623 if tempHost2 is None or len(notFoundCoords) > 0:

File ~/.conda/envs/torch2.0.0/lib/python3.9/site-packages/astro_ghost/ghostHelperFunctions.py:237, in getDBHostFromTransientCoords(transientCoords, GHOSTpath)
    234     notFound.append(transientCoord)
    235     continue
--> 237 c2 = SkyCoord(smallTable['TransientRA']*u.deg, smallTable['TransientDEC']*u.deg, frame='icrs')
    238 sep = np.array(transientCoord.separation(c2).arcsec)
    239 if np.nanmin(sep) <= 1:

File ~/.local/lib/python3.9/site-packages/astropy/coordinates/sky_coordinate.py:339, in SkyCoord.__init__(self, copy, *args, **kwargs)
    335 # Parse the args and kwargs to assemble a sanitized and validated
    336 # kwargs dict for initializing attributes for this object and for
    337 # creating the internal self._sky_coord_frame object
    338 args = list(args)  # Make it mutable
--> 339 skycoord_kwargs, components, info = _parse_coordinate_data(
    340     frame_cls(**frame_kwargs), args, kwargs
    341 )
    343 # In the above two parsing functions, these kwargs were identified
    344 # as valid frame attributes for *some* frame, but not the frame that
    345 # this SkyCoord will have. We keep these attributes as special
    346 # skycoord frame attributes:
    347 for attr in skycoord_kwargs:
    348     # Setting it will also validate it.

File ~/.local/lib/python3.9/site-packages/astropy/coordinates/sky_coordinate_parsers.py:313, in _parse_coordinate_data(frame, args, kwargs)
    309     for arg, frame_attr_name, repr_attr_name, unit in zip(
    310         args, frame_attr_names, repr_attr_names, units
    311     ):
    312         attr_class = frame.representation_type.attr_classes[repr_attr_name]
--> 313         _components[frame_attr_name] = attr_class(arg, unit=unit)
    315 else:
    316     raise ValueError(
    317         f"Must supply no more than three positional arguments, got {len(args)}"
    318     )

File ~/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py:667, in Longitude.__new__(cls, angle, unit, wrap_angle, **kwargs)
    663 if isinstance(angle, Latitude):
    664     raise TypeError(
    665         "A Longitude angle cannot be created from a Latitude angle."
    666     )
--> 667 self = super().__new__(cls, angle, unit=unit, **kwargs)
    668 if wrap_angle is None:
    669     wrap_angle = getattr(angle, "wrap_angle", self._default_wrap_angle)

File ~/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py:140, in Angle.__new__(cls, angle, unit, dtype, copy, **kwargs)
    135             angle = u.Quantity(angle, angle_unit, copy=False)
    137     elif isiterable(angle) and not (
    138         isinstance(angle, np.ndarray) and angle.dtype.kind not in "SUVO"
    139     ):
--> 140         angle = [Angle(x, unit, copy=False) for x in angle]
    142 return super().__new__(cls, angle, unit, dtype=dtype, copy=copy, **kwargs)

File ~/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py:140, in <listcomp>(.0)
    135             angle = u.Quantity(angle, angle_unit, copy=False)
    137     elif isiterable(angle) and not (
    138         isinstance(angle, np.ndarray) and angle.dtype.kind not in "SUVO"
    139     ):
--> 140         angle = [Angle(x, unit, copy=False) for x in angle]
    142 return super().__new__(cls, angle, unit, dtype=dtype, copy=copy, **kwargs)

File ~/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py:142, in Angle.__new__(cls, angle, unit, dtype, copy, **kwargs)
    137     elif isiterable(angle) and not (
    138         isinstance(angle, np.ndarray) and angle.dtype.kind not in "SUVO"
    139     ):
    140         angle = [Angle(x, unit, copy=False) for x in angle]
--> 142 return super().__new__(cls, angle, unit, dtype=dtype, copy=copy, **kwargs)

File ~/.local/lib/python3.9/site-packages/astropy/units/quantity.py:565, in Quantity.__new__(cls, value, unit, dtype, copy, order, subok, ndmin)
    562         cls = qcls
    564 value = value.view(cls)
--> 565 value._set_unit(value_unit)
    566 if unit is value_unit:
    567     return value

File ~/.local/lib/python3.9/site-packages/astropy/coordinates/angles.py:163, in Angle._set_unit(self, unit)
    162 def _set_unit(self, unit):
--> 163     super()._set_unit(self._convert_unit_to_angle_unit(unit))

File ~/.local/lib/python3.9/site-packages/astropy/units/quantity.py:2118, in SpecificTypeQuantity._set_unit(self, unit)
   2116 def _set_unit(self, unit):
   2117     if unit is None or not unit.is_equivalent(self._equivalent_unit):
-> 2118         raise UnitTypeError(
   2119             "{} instances require units equivalent to '{}'".format(
   2120                 type(self).__name__, self._equivalent_unit
   2121             )
   2122             + (
   2123                 ", but no unit was given."
   2124                 if unit is None
   2125                 else f", so cannot set it to '{unit}'."
   2126             )
   2127         )
   2129     super()._set_unit(unit)

UnitTypeError: Angle instances require units equivalent to 'rad', but no unit was given.

```
points at this line; https://github.com/uiucsn/astro_ghost/blob/main/astro_ghost/ghostHelperFunctions.py#LL236C103-L236C103 . I was able to progress by modifying by using .value to cast the series to np.ndarray

**Comment by alexandergagliano** [comment]
@awe2 Are you still running into this issue? I've just released a new version, v2.0.5, that shouldn't have these problems.

**Comment by awe2** [comment]
closing, v2.0.5 fixes the issue
